org.eclipse.m2e.core/src/org/eclipse/m2e/core/project/IProjectConfigurationManager.java

```diff
@@ -62,6 +63,7 @@ void enableMavenNature(IProject project, IProjectConfiguration configuration, IP
 
   void updateProjectConfiguration(IProject project, IProgressMonitor monitor) throws CoreException;
 
+  @Deprecated(forRemoval = true)
```

Comment:
Is there an alternative users should use? I think it would be good to add one or if not make clear why it should not be used and what should be done instead.